src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs

```diff
@@ -158,10 +158,10 @@ public void DeeplyNestedGeneric()
                 (ExecutionArchitecture.x64, ExecutionConfiguration.Release) when ExecutionConditionUtil.IsMacOS => 520,
                 _ when ExecutionConditionUtil.IsCoreClrUnix => 1200,
                 _ when ExecutionConditionUtil.IsMonoDesktop => 730,
-                (ExecutionArchitecture.x86, ExecutionConfiguration.Debug) => 520,
-                (ExecutionArchitecture.x86, ExecutionConfiguration.Release) => 1640,
-                (ExecutionArchitecture.x64, ExecutionConfiguration.Debug) => 290,
-                (ExecutionArchitecture.x64, ExecutionConfiguration.Release) => 810,
+                (ExecutionArchitecture.x86, ExecutionConfiguration.Debug) => 460,
```

Comment:
What is the reference environment to run this test?

Reply:
Running xunit.console.exe or xunit.console.x86.exe on my home PC.

Reply:
Well, that is how I determined new baselines. Perhaps we could do it in a more "canonical" environment, e.g. by lowering all the baselines, uncommenting the bits that find the new baseline, then running it on a CI machine.

Reply:
The numbers I've set were the test run on Azure VM (VS 2019 Latest image) using `./eng/cibuild.ps1`. I thought those would be a "canonical" enough method to measure baselines. 😄
Also, what's interesting, running the commented loop leads to different number than running the test itself trying to adjust the number. Maybe will check why it is so later.

Yeah, and the question about Linux environment is still actual. Do we want to adjust the number for it? The real issue there is the test duration which reaches 40 minutes.

Reply:
IMO, there is no need to test the actual limit on Linux. Keeping at around the desktop Release/x86 level or a bit higher is fine.

Reply:
BTW, the numbers mentioned in the PR description were measured by running `dotnet test`.

Reply:
I also noticed that the thresholds in the test differ based on simply whether you use the loop or not. It is a bit of a pain.

The most likely reason your thresholds differed from mine is that some features have gone into master since this PR was opened, most notably records.

Reply:
Interesting enough, they differ in different direction. For macOS the working threshold was lower then the one measured using the loop and for Windows it's vice versa.